gammapy/time/models.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import logging
+import numpy as np
+from ..utils.modeling import Parameter, ParameterList
+
+__all__ = [
+    'PhaseCurve',
+]
+
+log = logging.getLogger(__name__)
+
+
+class PhaseCurve(object):
+    """PhaseCurve class.
+    
+    This is to calculate phase of a periodic system and to provide the 
+    normalization factor for the phase using circular interpolation. 
+    The required data for interpolation is given as an astropy table
+
+    # TODO: 1. Checking table values if the phases are in increasing order
+            2. Check normalization constant to set maximum values
+
+    Parameters
+    ----------
+    table = A table of 'PHASE' vs 'NORM' should be given
+    reference: The MJD value where phase is considered as 0.
+    phase0: phase at the reference MJD
+    f0:1st order derivative of the function phi with time
+    f1:2nd order derivative of the function phi with time
+    f2:3rd order derivative of the function phi with time
+    phase as function of time is computed using
+    Φ(t)=Φ0+f0(t−t0)+(1/2)f1(t−t0)^2+(1/6)f2(t−t0)^3
+
+
+    Returns
```

Comment:
The class docstring should not have a returns section.
It has a Parameters section that documents the parameters of `__init__`.
`__init__` always returns an instance, doesn't contain a return statement and doesn't need a documentation of the return value.
Presumably here you're documenting the return value of one of the methods. That should be in the docstring of that method.